core/templates/pages/exploration-player-page/learner-experience/tutor-card.component.ts

```diff
@@ -271,6 +272,39 @@ export class TutorCardComponent {
     return this.translateService.instant(milestoneMessageTranslationKey);
   }
 
+  setNextMilestoneAndCheckIfProgressBarIsShown(): boolean {
+    if (
+      !this.inStoryMode ||
+      this.completedChaptersCount > 50 ||
+      (this.milestoneMessageIsToBeDisplayed &&
+        MILESTONE_SPECIFIC_COMPLETED_CHAPTER_COUNTS.includes(
+          this.completedChaptersCount))
+    ) {
```

Comment:
Can you put this condition into a function with some name for better readability.

Reply:
Done.